src/resource/archive/O2rArchive.cpp

```diff
@@ -10,6 +10,7 @@ O2rArchive::O2rArchive(const std::string& archivePath) : Archive(archivePath) {
 
 O2rArchive::~O2rArchive() {
     SPDLOG_TRACE("destruct o2rarchive: {}", GetPath());
+    Close();
```

Comment:
`Close` just has

```cpp
bool O2rArchive::Close() {
    if (zip_close(mZipArchive) == -1) {
        SPDLOG_ERROR("Failed to close zip file \"{}\"", GetPath());
        return false;
    }
    return true;
}
```

i think we should probably add a null check for `mZipArchive` into close so we don't get failed to close errors if we never opened a zip file

Reply:
hmm if mZipArchive are null then I guess zip_close will return -1, I will double check that

Reply:
try it and yeah it's the case.